woocommerce/woocommerce-ios
# Plugins: Hide plugins from settings from non-admin roles

closes #4276

# Why 

Managing plugins is only available for admin roles, when @itsmeichigo was working on listing plugins for her trial, we had the idea to restrict the app only to admin roles, however, that decision changed and this PR **hides the plugin list from non-admin roles and removes the feature flag the feature was behind.**

# How

To accomplish this I'm taking advantage of the work recently introduced in #4492, which already fetches the role of the user at start-up or after a store switch and saving those roles in the `SessionManager`. 

I spent a while thinking if I should store those roles in `CoreData` using the `User` entity but I finally decided against it because most(or all?) "default store" information is already persisted in the session manager.

Finally, `SettingsViewControllers` fetches the session manager information to know if it should display the plugins section or not.

# Screenshots

Shop Manager | Admin
---- | ----
<img width="384" alt="no-plugins" src="https://user-images.githubusercontent.com/562080/128361914-ad88d553-a6e0-457d-a546-f6e5cfc68e50.png"> | <img width="380" alt="plugins" src="https://user-images.githubusercontent.com/562080/128361906-6367833f-1ad9-4cb0-a05c-cad19e591fb1.png">


# Testing Steps

- Launch the app with an admin user(make sure you are able to change the user's role).
- Navigate to the settings screen and check that you see the plugins section.
- In Core change the user role to "shop manager"
- Relaunch the app
- Navigate to the settings screen and check that you don't see the plugins section.


Update release notes:
